File: samza-core/src/main/scala/org/apache/samza/config/loaders/PropertiesConfigLoader.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

package org.apache.samza.config.loaders

import java.io.FileInputStream
import java.util.Properties

import scala.collection.JavaConverters._
import org.apache.samza.SamzaException
import org.apache.samza.config.{Config, ConfigLoader, MapConfig}
import org.apache.samza.util.Logging

/**
 * Load config from a properties file with given path. "path" will be given in the input properties map.
 */
class PropertiesConfigLoader extends ConfigLoader with Logging {

  def getConfig(properties: java.util.Map[String, String]): Config = {
    val path = properties.get("path")
    if (path == null) {
      throw new SamzaException("path is required to read config from properties file")
    }

    val in = new FileInputStream(path)
    val props = new Properties()

    props.load(in)
    in.close()

    debug("got config %s from path %s" format (props, path))

    new MapConfig(props.asScala.asJava)
  }
}

File: samza-core/src/main/java/org/apache/samza/config/loaders/PropertiesConfigLoader.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

package org.apache.samza.config.loaders;

import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.util.HashMap;
import java.util.Map;
import java.util.Properties;
import org.apache.samza.SamzaException;
import org.apache.samza.config.Config;
import org.apache.samza.config.ConfigLoader;
import org.apache.samza.config.MapConfig;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import static java.util.Objects.requireNonNull;


/**
 * ConfigLoader to load full job configs from a local properties file given its path.
 */
public class PropertiesConfigLoader implements ConfigLoader {
  private static final Logger LOG = LoggerFactory.getLogger(PropertiesConfigLoader.class);

  private final String path;

  public PropertiesConfigLoader(String path) {
    this.path = requireNonNull(path);
  }

  @Override
  public Config getConfig() {
    try {
      InputStream in = new FileInputStream(path);
      Properties props = new Properties();

      props.load(in);
      in.close();

      LOG.debug("got config {} from path {}", props, path);

      Map<String, String> config = new HashMap<>(props.size());
      // Per Properties JavaDoc, all its keys and values are of type String
      props.forEach((key, value) -> config.put(key.toString(), value.toString()));

      return new MapConfig(config);
    } catch (IOException e) {
      throw new SamzaException("Failed to read from " + path);
    }
  }
}